src/libCrypto/EthCrypto.cpp

```diff
@@ -435,3 +504,18 @@ bytes StripEVM(bytes const& in) {
     return in;
   }
 }
+
+bytes CreateHash(std::string const& rawTx) {
+  auto const asBytes = DataConversion::HexStrToUint8VecRet(rawTx);
+
+  auto const hash = ethash::keccak256(asBytes.data(), asBytes.size());
+
+  bytes hashBytes;
+
+  hashBytes.insert(hashBytes.end(), &hash.bytes[0], &hash.bytes[32]);
+
+  auto asString = DataConversion::Uint8VecToHexStrRet(hashBytes);
+  boost::algorithm::to_lower(asString);
```

Comment:
Why do you need this lowercase string here if hashBytes are then returned.

Reply:
good spot! that was just used for debug and can be deleted.